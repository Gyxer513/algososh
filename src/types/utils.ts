import { ElementStates } from "./element-states";

export interface IItemArray {
  item: string | number;
  state: ElementStates;
}

export interface ISortArray {
  state: number;
  color: ElementStates;
}


export enum SortMethod {
  Select = "Select",
  Bubble = "Bubble",
}

export interface IStackElement {
  item: string;
  state: ElementStates;
}

export interface IQueueElement {
  item: string;
  state: ElementStates;
}