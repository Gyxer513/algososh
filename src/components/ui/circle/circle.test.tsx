import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Circle tests", () => {
  /* без буквы */
  it("Circle without letter", () => {
    const circle = <Circle />;
    render(circle);
    expect(screen.getByTestId("circleContent")).toHaveTextContent("");
  });
  /* с буквами */
  it("Circle with letter", () => {
    const circle = <Circle letter="a" />;
    render(circle);
    expect(screen.getByTestId("circleContent")).toHaveTextContent("a");
  });
  /* с head */
  it("Circle with head", () => {
    const circle = <Circle head="a" />;
    render(circle);
    expect(screen.getByTestId("head")).toHaveTextContent("a");
  });
  /* с react-элементом в head */
  it("Circle with circle head", () => {
    const circle = <Circle tail={<Circle />} />;
    render(circle);
    expect(screen.getAllByTestId("circleContent").length).toBe(2);
  });
  /* с tail */
  it("Circle with tail", () => {
    const circle = <Circle tail="a" />;
    render(circle);
    expect(screen.getByTestId("tail")).toHaveTextContent("a");
  });
  /* с react-элементом в tail */
  it("Circle with tail head", () => {
    const circle = <Circle tail={<Circle />} />;
    render(circle);
    expect(screen.getAllByTestId("circleContent").length).toBe(2);
  });
  /* c index */
  it("Circle with index", () => {
    const circle = <Circle index={0} />;
    render(circle);
    expect(screen.getByTestId("circleContent")).toHaveTextContent(String(0));
  });
  /* с пропом isSmall ===  true */
  it("Circle is small", () => {
    const circle = <Circle isSmall={true} />;
    render(circle);
    screen.debug();
    expect(screen.getByTestId("circle")).not.toHaveStyle("width: 80px");
  });
  /* в состоянии default */
  it("Circle Default", () => {
    const circle = <Circle state={ElementStates.Default} />;
    render(circle);
    expect(screen.getByTestId("circle")).toHaveStyle(
      "border-color:  --default-color"
    );
  });
  /* в состоянии changing */
  it("Circle Changing", () => {
    const circle = <Circle state={ElementStates.Changing} />;
    render(circle);
    expect(screen.getByTestId("circle")).toHaveStyle(
      "border-color:  --changing-color"
    );
  });
  /*  в состоянии modified */
  it("Circle Modified", () => {
    const circle = <Circle state={ElementStates.Modified} />;
    render(circle);
    expect(screen.getByTestId("circle")).toHaveStyle(
      "border-color:  --modified-color"
    );
  });
});
