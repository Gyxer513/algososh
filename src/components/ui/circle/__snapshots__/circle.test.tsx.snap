// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle tests Circle Changing 1`] = `
<Circle
  state="changing"
/>
`;

exports[`Circle tests Circle Default 1`] = `
<Circle
  state="default"
/>
`;

exports[`Circle tests Circle Modified 1`] = `
<Circle
  state="modified"
/>
`;

exports[`Circle tests Circle is small 1`] = `
<Circle
  isSmall={true}
/>
`;

exports[`Circle tests Circle with circle head 1`] = `
<Circle
  tail={<Circle />}
/>
`;

exports[`Circle tests Circle with head 1`] = `
<Circle
  head="a"
/>
`;

exports[`Circle tests Circle with index 1`] = `
<Circle
  index={0}
/>
`;

exports[`Circle tests Circle with letter 1`] = `
<Circle
  letter="a"
/>
`;

exports[`Circle tests Circle with tail 1`] = `
<Circle
  tail="a"
/>
`;

exports[`Circle tests Circle with tail head 1`] = `
<Circle
  tail={<Circle />}
/>
`;

exports[`Circle tests Circle without letter 1`] = `<Circle />`;
